ape_infura/provider.py

```diff
@@ -34,6 +35,23 @@ def __init__(self):
 
 class Infura(Web3Provider, UpstreamProvider):
     network_uris: dict[tuple[str, str], str] = {}
+    api_keys: list[str] = []
```

Comment:
using a set is better here

```suggestion
    api_keys: set[str] = []
```